pandas/tests/test_sorting.py

```diff
@@ -413,14 +413,16 @@ def test_mixed_integer_from_list(self):
         expected = np.array([0, 0, 1, 'a', 'b', 'b'], dtype=object)
         tm.assert_numpy_array_equal(result, expected)
 
-    @pytest.mark.skipif(PY2, reason="pytest.raises match regex fails")
     def test_unsortable(self):
         # GH 13714
         arr = np.array([1, 2, datetime.now(), 0, 3], dtype=object)
-        msg = (r"'(<|>)' not supported between instances of ('"
-               r"datetime\.datetime' and 'int'|'int' and 'datetime\.datetime"
-               r"')|"
-               r"unorderable types: int\(\) > datetime\.datetime\(\)")
+        msg = (r"unorderable types: ({0} [<>] {1}|{1} [<>] {0})".format(
```

Comment:
can you explain why *anything* needs to change except for the CI files. The point of change is NOT to change anything in test files.

Reply:
It's not that hard: the CI would fail without this change, because the current regex is just good enough for the current CI, but not good enough once python/numpy versions change.